spec/unit/indirector/rest_spec.rb

```diff
@@ -353,7 +353,7 @@ def save_request(key, instance, options={})
       it "calls get on the connection" do
         request = find_request('foo bar')
 
-        connection.expects(:get).with("#{url_prefix}/test_model/foo%20bar?environment=production&", anything).returns(mock_response('200', 'response body'))
+        connection.expects(:get).with("#{url_prefix}/test_model/foo+bar?environment=production&", anything).returns(mock_response('200', 'response body'))
```

Comment:
It looks like puppetserver/jetty does not decode `+`. It does decode `%20` as space.

I tested this on master branch of Puppet Server with both this PR and without (with cef0271da01a4212d6b5ab589cd10f8035efa768 of Puppet, which was HEAD last week) by doing `curl -k https://localhost:8140/puppet/v3/node/foo%20bar?environment=production` and `curl -k https://localhost:8140/puppet/v3/node/foo+bar?environment=production`.

With `foo%20bar` I got `{"name":"foo bar","environment":"production"}` as the response, and saw in the logs:

```
2017-04-04 11:30:20,267 DEBUG [qtp541735488-33] [p.r.core] Processing :get /puppet/v3/node/foo bar
...
2017-04-04 11:30:20,470 DEBUG [qtp541735488-33] [puppetserver] Puppet Did not match path ("/node/foo bar")
...
2017-04-04 11:30:20,500 INFO  [qtp541735488-33] [puppetserver] Puppet Caching node for foo bar
```

With `foo+bar` I got `{"name":"foo+bar","environment":"production"}` as the response, and saw in the logs:

```
2017-04-04 11:30:31,088 DEBUG [qtp541735488-33] [p.r.core] Processing :get /puppet/v3/node/foo+bar
...
2017-04-04 11:30:31,091 DEBUG [qtp541735488-33] [puppetserver] Puppet Did not match path ("/node/foo+bar")
...
2017-04-04 11:30:31,096 INFO  [qtp541735488-33] [puppetserver] Puppet Caching node for foo+bar
```

Given that I got the same results with and without this PR, it does seem like this should be backwards compatible though.